thorlcr/activities/funnel/thfunnelslave.cpp

```diff
@@ -795,20 +794,36 @@ class CNWaySelectActivity : public CSlaveActivity, public CThorSteppable
                 IThorNWayInput *nWayInput = dynamic_cast<IThorNWayInput *>(cur);
                 if (nWayInput)
                 {
+                    cur->start();
                     unsigned numRealInputs = nWayInput->numConcreteOutputs();
                     if (whichInput < numRealInputs)
                     {
                         selectedInputITDL = nWayInput->queryConcreteInput(whichInput);
-                        selectedStream = connectSingleStream(*this, selectedInputITDL, 0, selectedJunction, true);  // Should this be passing whichInput??
+                        selectedStream = nWayInput->queryConcreteInputStream(whichInput);
+                        selectedJunction = nWayInput->queryConcreteInputJunction(whichInput);
                         break;
                     }
                     whichInput -= numRealInputs;
                 }
+                else
+                {
+                    if (whichInput == 0)
+                    {
+                        selectedInputITDL = cur;
```

Comment:
It will be started by the : 
```
if (selectedInputITDL)
    selectedInputITDL->start()
```
.. block below.

It's only a NWay input that needs (in addition) to be started here.